### PR TITLE
Feature/bloom sanity seo settings

### LINKED ISSUE #1225: Configure Sanity-controlled SEO settings
### Link task to story
Story: #1214

### Description
Configure Sanity-controlled SEO settings:
- Sitemap.
- Redirects.

### PR DESCRIPTION
**Pull request by jkellerman**
## ☑ List all github issues that this PR will close:

- Closes #1225 

## 📝 Description

This adds the SEO settings for bloom connections site.  A lot of setup has already been taken care of in osc-studio, so it was case of making sure the following is included in the recruitment app:

- Making sure groq queries for pages include SEO
- Creating canonical url and html meta tag functions and inserting them into our loader for each page route.

## ⛳️ Current behaviour (updates)

N/A

## 🚀 New behaviour

N/A

## 💣 Is this a breaking change (Yes/No): No

> Please describe any breaking changes.

## 🧾 Final Checklist

- [ ] I've considered possible cross-platform impacts (Mac, Linux, Windows)
- [ ] I've considered possible cross-device impacts (iOS, Android)
- [ ] I've considered possible cross-browser impacts (Chrome, Safari, Edge, Firefox)
- [ ] I've added tests to cover my changes (Unit, E2E, Integration, Contract, Functional, Acceptance etc)
- [ ] I've added or updated any documentation.

## 📝 Additional Information

> Any additional contextual information, such as relevant links/articles/documentation

### Summary by CodeRabbit

**New Features:**
- Added SEO-related data to the home page, job page, jobs landing page, and other pages for improved search engine optimization.
- Introduced Google Tag Manager and HubSpot embed code in the `App` component for enhanced analytics tracking.
- Implemented a new `meta` function that generates metadata for SEO and social media sharing.
- Created utility functions `buildCanonicalUrl` and `buildHtmlMetaTags` for generating canonical URLs and HTML meta tags respectively.

**Refactor:**
- Updated GraphQL queries to include SEO information and analytics IDs.
- Modified JSX expressions to check for Google Tag Manager ID instead of HubSpot ID.

**Chore:**
- Disabled the v2 meta feature in the `remix.config.js` file.

### CODE REVIEW

File: packages/osc-recruitment/app/root.tsx [hunk 1]
```diff
@@ -100,6 +100,41 @@ export const loader: LoaderFunction = async ({ request, context }) => {
...
 
+export const meta: MetaFunction = ({ data }) => {
+    const { seo: seoSettings } = data.siteSettings;
+
+    const noindex = seoSettings?.robots?.noIndex && 'noindex';
+    const organizationAsset = seoSettings?.schema?.organizationLogo.asset;
+    const facebook = seoSettings.socials.find((social: { socialProfile: string }) =>
+        social.socialProfile.includes('facebook')
+    );
+    console.log('facebook', facebook);
```
Comment (stevenbister) [inline packages/osc-recruitment/app/root.tsx:111]: Can you remove any console logs please

File: packages/osc-recruitment/app/root.tsx [hunk 2]
```diff
@@ -100,6 +100,41 @@ export const loader: LoaderFunction = async ({ request, context }) => {
...
+    const twitter = seoSettings.socials.find((social: { socialProfile: string }) =>
+        social.socialProfile.includes('twitter')
+    );
+
+    const twitterHandle =
+        twitter && twitter.socialProfile.substring(twitter.socialProfile.lastIndexOf('/') + 1);
+
+    return {
+        charset: 'utf-8',
```
Comment (stevenbister) [inline packages/osc-recruitment/app/root.tsx:120]: Looks like you've already got this set elsewhere so you can remove this as we're getting duplicates

File: packages/osc-recruitment/app/root.tsx [hunk 3]
```diff
@@ -100,6 +100,41 @@ export const loader: LoaderFunction = async ({ request, context }) => {
...
+        title: seoSettings?.siteTitle,
+        viewport: 'width=device-width,initial-scale=1',
```
Comment (stevenbister) [inline packages/osc-recruitment/app/root.tsx:122]: Looks like you've already got this set elsewhere so you can remove this as we're getting duplicates

File: packages/osc-recruitment/app/root.tsx [hunk 4]
```diff
@@ -114,6 +149,20 @@ export default function App() {
                 <DynamicLinks />
             </head>
             <body className="theme theme--bloom">
+                {/* Google Tag Manager (noscript) */}
+                {process.env.NODE_ENV === 'development' ||
+                !siteSettings.analytics.hubspotId ? null : (
```
Comment (stevenbister) [inline packages/osc-recruitment/app/root.tsx:154]: This is currently checking for the hubspot id not the google tag manager one.

Also Google tag manager requires a tag in the `<head>` to work as well as this one in the body. Think there's an example in osc-ecommerce

File: packages/osc-recruitment/app/routes/index.tsx [hunk 5]
```diff
@@ -26,20 +28,32 @@ export const loader: LoaderFunction = async ({ request, params }) => {
...
+        canonicalUrl,
         isPreview,
         query: isPreview ? HOME_QUERY : null,
         params: isPreview ? params : null,
     });
 };
 
+export const meta: MetaFunction = ({ data, parentsData }) => {
+    const globalSeoSettings = parentsData.root.siteSettings.seo;
+    console.log('globalSeoSettings', globalSeoSettings);
```
Comment (stevenbister) [inline packages/osc-recruitment/app/routes/index.tsx:47]: Can you remove any console logs please

**Review by stevenbister — CHANGES_REQUESTED** [review]
Good stuff, couple of comments for you

File: packages/osc-recruitment/app/root.tsx (hunk 4 above)
Comment (jkellerman) [reply packages/osc-recruitment/app/root.tsx:154]: Yeah sure I have added the tag in the `<head>`.

Regarding it currently checking for hubspot id, it is also doing this in ecommerce.

Comment (stevenbister) [reply packages/osc-recruitment/app/root.tsx:163]: Ahh okay are you okay to update both? Should match be `gtmId` rather than `hubspotId`

Comment (jkellerman) [reply packages/osc-recruitment/app/root.tsx]: That's done

**Review by stevenbister — CHANGES_REQUESTED** [review]
Sorry mate, just noticed that https://pr-1583-osc-recruitment.fly.dev/ is erroring

this is the stack trace in fly
```
2023-09-08T07:29:30.691 app[683d477a7560d8] lhr [info] [Error: Unexpected Server Error]

2023-09-08T07:29:30.699 app[683d477a7560d8] lhr [info] GET / 500 - - 725.371 ms

2023-09-08T07:29:33.478 app[683d477a7560d8] lhr [info] params undefined

2023-09-08T07:29:33.767 app[683d477a7560d8] lhr [info] TypeError: Cannot read properties of undefined (reading 'gtmId')

2023-09-08T07:29:33.767 app[683d477a7560d8] lhr [info] at App (/app/packages/osc-recruitment/build/index.js:633:30)

2023-09-08T07:29:33.767 app[683d477a7560d8] lhr [info] at Uc (/app/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:70:44)

2023-09-08T07:29:33.767 app[683d477a7560d8] lhr [info] at Xc (/app/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:72:253)

2023-09-08T07:29:33.767 app[683d477a7560d8] lhr [info] at Z (/app/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:78:89)

2023-09-08T07:29:33.767 app[683d477a7560d8] lhr [info] at Xc (/app/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:72:481)

2023-09-08T07:29:33.767 app[683d477a7560d8] lhr [info] at Z (/app/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:78:89)

2023-09-08T07:29:33.767 app[683d477a7560d8] lhr [info] at Xc (/app/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:76:231)

2023-09-08T07:29:33.767 app[683d477a7560d8] lhr [info] at Z (/app/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:78:89)

2023-09-08T07:29:33.767 app[683d477a7560d8] lhr [info] at Xc (/app/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:72:481)

2023-09-08T07:29:33.767 app[683d477a7560d8] lhr [info] at Z (/app/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:78:89)
```

`gtmId` is undefined so looks like you will need to add it to your settings query for bloom

**Review by stevenbister — APPROVED** [review]
Good stuff!

**Pull request merged**